flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamingRuntimeContext.java

```diff
@@ -173,4 +174,12 @@ public void registerTimer(long time, Triggerable target) {
 			}
 		}
 	}
+
+	/**
+	 * Returns true if checkpointing is enabled for the running job.
+	 * @return true if checkpointing is enabled.
+	 */
+	public boolean isCheckpointingEnabled() {
+		return taskEnvironment.getTaskConfiguration().getBoolean(StreamConfig.CHECKPOINTING_ENABLED, false);
```

Comment:
That breaks the way that config values are accessed. The right way is
`new StreamConfig(taskEnvironment.getTaskConfiguration()).isCheckpointingEnabled()`.

Then you also need not change the visibility of the flag.